modules/extensions/src/test/java/org/apache/synapse/mediators/bsf/javascript/JavaScriptMediatorTest.java

```diff
@@ -63,4 +64,45 @@ public void testInlineMediatorWithImports() throws Exception {
         boolean response = mediator.mediate(mc);
         assertTrue(response);
     }
+
+    /**
+     * Test controlling access to java classes through JS
+     * @throws Exception
+     */
+    public void testJavaClassAccessControl() throws Exception {
+        String scriptSourceCode =  "var s = new java.util.ArrayList();\n";
+
+
+        MessageContext mc = TestUtils.getTestContext("<foo/>", null);
+        ScriptMediator mediator = new ScriptMediator("js", scriptSourceCode, null);
+
+        System.setProperty("properties.file.path", System.getProperty("user.dir") + "/src/test/resources/file.properties");
+
+        try {
+            mediator.mediate(mc);
+            fail("Failed to enforce Java class access control configuration during mediation");
+        } catch(SynapseException e) {}
```

Comment:
Shall we assert for true in the catch block, since it's the excepted code block to get hit?

Reply:
Fixed in 
f7bd88e